trio/__init__.py

```diff
@@ -13,61 +13,81 @@
 # This file pulls together the friendly public API, by re-exporting the more
 # innocuous bits of the _core API + the higher-level tools from trio/*.py.
 
-from ._version import __version__
+__all__ = [
+    "TrioInternalError",
+    "RunFinishedError",
+    "WouldBlock",
+    "Cancelled",
+    "ResourceBusyError",
+    "ClosedResourceError",
+    "MultiError",
+    "run",
+    "open_nursery",
+    "open_cancel_scope",
+    "current_effective_deadline",
+    "TASK_STATUS_IGNORED",
+    "current_time",
+    "TaskLocal",
+]
```

Comment:
So we don't actually need `__all__` at all :-)

If you do `from module import *`, then Python:

1. Checks for an `__all__`. If it finds one, then that's what it uses.
2. Otherwise, it imports all names that are defined in the module, and don't start with an underscore.

So we can (and should) delete this `__all__` entirely. So long as we're careful about what names we export – which is something we need to do anyway – then it's totally redundant.